module/VuFindTheme/src/VuFindTheme/View/Helper/SetupThemeResources.php

```diff
@@ -141,21 +140,35 @@ protected function addLinks()
     }
 
     /**
-     * Add scripts to header.
+     * Add scripts to header or footer.
      *
-     * @return void
+     * @return null
      */
     protected function addScripts()
     {
+        $legalHelpers = ['footScript', 'headScript'];
+
         // Load Javascript (same ordering considerations as CSS, above):
         $js = array_reverse($this->container->getJs());
-        $headScript = $this->getView()->plugin('headScript');
+
         foreach ($js as $current) {
-            $headScript()->forcePrependFile(
-                $current['file'],
-                'text/javascript',
-                $current['attributes'] ?? []
-            );
+            $position = $current['position'] ?? 'header';
+            $helper = substr($position, 0, 4) . 'Script';
+            if (!in_array($helper, $legalHelpers)) {
+                throw new \Exception(
+                    'Invalid script position for '
+                    . $current['file'] . ': ' . $position . '.'
+                );
+            }
+
+            $this->getView()
+                ->plugin($helper)
+                ->forcePrependFile(
+                    $current['file'],
+                    'text/javascript',
+                    $current['attributes'] ?? []
+                );
         }
+        return null;
```

Comment:
Is there a reason for this to return null instead of remaining a void method? @crhallberg, did you do this for a reason, or was it already here?

Reply:
Already there from when footer scripts were returned as an inline string. Void would be fine.

Reply:
Fixed, thanks!